src/tools/base/segmentation/BaseBrushSegmentationTool.js

```diff
@@ -11,7 +11,7 @@ const { state, getters, setters } = store.modules.brush;
 
 /**
  * @abstract
- * @memberof Tools.Base
+ * @memberof Tools.BaseSegmentationTool
```

Comment:
From the doc it was my understanding that `BaseBrushSegmentationTool` should extend `BaseTool`, right @erik ? Tbh the naming is little confusing.

The idea is that there are two types of segmentations tools that share common attributes:
- Tools where have a circular cursor and draw on the canvas to produce a segmentation (i.e. BaseBrushSegmentationTool`).
- Tools where you select an area and some processing event happens (.e.g BaseSegmentationTool).

Perhaps leaving it as `BaseBrushTool` is actually less confusing? I'm not sure.

Both of these should extend `BaseTool`, I believe.

Reply:
Hmm, I've seen what you've done here, You've made BaseSegmentationTool a very light layer (different to what we intended in the spec).

I know I'm hitting in early when this is WIP, but architecturally I would like to get this right first time. The current implementation of `BaseSegmentationTool` looks like it should be a mixin, I'll get back to you after having a think.